### DISCUSSION TITLE
Woche 3

### DISCUSSION THREAD

**Discussion by TimSchnur** [Christian-2003/CrInGE#9] [Student Exchange]
Hallo zusammen,

willkommen zu unserem dritten Blogeintrag.

Diese Woche haben wir uns mit der Ausarbeitung unserer :page_facing_up: **Softwareanforderungen** beschäftigt.
Für die [Software Anforderungsspezifikation](https://github.com/Christian-2003/CrInGE/blob/master/docs/SoftwareAnforderungsspezifikation.md) haben wir einen eigenen :file_folder: Ordner für solche Dokumente angelegt, uns eine Struktur für diesen Ordner überlegt und mit einer [Read me](https://github.com/Christian-2003/CrInGE/blob/master/docs/README.md) versehen.

###### Aufgabenverteilung
Für die unterschiedlichen Aufgaben haben wir uns folgendermaßen aufgeteilt:
| Aufgabe | Programm | Verantwortliche |
|--------|--------|--------|
| SRS designen :bookmark_tabs: | Markdown | @Christian-2003 |
| User Stories :busts_in_silhouette: | GitHub Projects | @TimSchnur |
| UI Mockups :computer: | MS Visio | @Flinks271, @FabianDev001, @Elekt0 |
| UML Diagramme :chart_with_upwards_trend: | Plant UML | @Flinks271, @FabianDev001, @Elekt0 | 

###### Sonstiges
Des Weiteren haben wir mit der Implementierung der :octocat: **GitHub Actions** begonnen.
Diese sollen die Anwendung nicht nur Builden und Testen, sondern auch Releases automatisch veröffentlichen.

Außerdem wurden 🔍 Unittests zu den bereits vorhanden Klassen geschrieben.

Zudem haben wir einige Prinzipen für unsere 🔜 **künftige Arbeitsweise** festgelegt:
* 🔎 **Test Driven Development** zur Qualitätssicherung
* 📝 **Java Docs** zur Code Dokumentation
* 🔑 **Feature Branches** zur Repository Konsistenz

**Comment by lukpr0** [comment]
Eure SRS ist ausführlich geschrieben, gefällt mir. Fände es jedoch besser, wenn  ihr so etwas wie UI Mockups und UML-Diagramme direkt im Blog referenziert.

**Reply by Christian-2003** [reply]
Hallo,

vielen Dank für dein Feedback, es freut uns sehr dass dir unser SRS gefällt.

Bezüglich der Referenzierung der UI-Mockups umd UML-Diagramme wird das SRS aktuell überarbeitet. Das sollte dann bis zur nächsten Vorlesung erledigt sein. Gerne kannst du uns im nächsten Blog mitteilen, ob dir das überarbeitete SRS gefällt. Wir freuen uns diesbezüglich über jedes Feedback.
